### ISSUE TITLE
Support dashed lines in PathLayer

### ISSUE THREAD

**Issue by Pessimistress** [uber/deck.gl#398]
API proposal:

#### `getStrokeDashArray` (Function, optional)

Returns the dash array for each path, as an array of two numbers 
`[solid_stroke_length, gap_length]`. Length is relative to stroke width.

#### `getStrokeDashOffset` (Function, optional)

Returns the dash offset for each path, as a numeric value.
The dash offset shifts the solid strokes along the length of the path.
Relative to stroke width.

#### `justified` (bool, optional)

- Default: `true`

Whether the alignment of the dashes are adjusted to fit the length of the path.
false - no adjustment, visible seams at joints
```
o----     ----     ----     ---- o----     -o----     ----     o
```
true - gap lengths may be stretched/compressed so that the dash fits perfectly between the two
ends. Half solid stroke length is drawn at each end for nicer joints.
```
o--    ----    ----    ----    --o--      --o--     ----     --o
```

@ibgreen

**Comment by ibgreen** [comment]
@Pessimistress I personally think this feature is amazing and we should add it to 4.2. If you disagree, please close the ticket.

**Comment by ibgreen** [comment]
Closed by #898
